src/climate_learn/models/modules/forecast.py

```diff
@@ -215,6 +215,134 @@ def test_step(self, batch: Any, batch_idx: int):
         days = [int(s * pred_range / 24) for s in steps]
         day = int(days_each_step)
 
+        # rmse for climatology baseline
+        clim_pred = self.train_clim  # C, H, W
+        clim_pred = (
+            clim_pred.unsqueeze(0)
+            .unsqueeze(0)
+            .repeat(y.shape[0], y.shape[1], 1, 1, 1)
+            .to(y.device)
+        )
+        baseline_rmse = lat_weighted_rmse(
+            clim_pred,
+            y,
+            out_variables,
+            transform_pred=False,
+            transform=self.denormalization,
+            lat=self.lat,
+            log_steps=steps,
+            log_days=days,
+        )
+        for var in baseline_rmse.keys():
+            self.log(
+                "test_climatology_baseline/" + var,
+                baseline_rmse[var],
+                on_step=False,
+                on_epoch=True,
+                sync_dist=True,
+                batch_size=len(x),
+            )
+
+        # rmse for persistence baseline
+        pers_pred = x  # B, 1, C, H, W
+        baseline_rmse = lat_weighted_rmse(
+            pers_pred,
+            y,
+            out_variables,
+            transform_pred=True,
+            transform=self.denormalization,
+            lat=self.lat,
+            log_steps=steps,
+            log_days=days,
+        )
+        for var in baseline_rmse.keys():
+            self.log(
+                "test_persistence_baseline/" + var,
+                baseline_rmse[var],
+                on_step=False,
+                on_epoch=True,
+                sync_dist=True,
+                batch_size=len(x),
+            )
+
+        # rmse for linear regression baseline
+        # check if fit_lin_reg_baseline is called by checking whether self.lr_baseline is initialized
+        try:
+            lr_pred = self.lr_baseline.predict(
+                x.cpu().reshape((x.shape[0], -1))
+            ).reshape(y.shape)
+        except AttributeError as e:
```

Comment:
I suspect the error is here. Lines 270-300 or so should be wrapped in an if-block.

Reply:
good catch, i made the change